x-pack/plugins/ml/public/alerting/ml_anomaly_alert_trigger.tsx

```diff
@@ -53,6 +59,39 @@ const MlAnomalyAlertTrigger: FC<MlAnomalyAlertTriggerProps> = ({
     []
   );
 
+  const jobsAndGroupIds: string[] = useMemo(
+    () => (Object.values(alertParams.jobSelection ?? {}) as string[][]).flat(),
+    [alertParams.jobSelection]
+  );
+
+  /**
+   * Extract alert related information based on the job selection
+   */
+  const fetchJobsConfig = useCallback(async () => {
+    try {
+      const jobs = await adJobsApiService.jobs(jobsAndGroupIds);
```

Comment:
if all you're after here is the influencer lists, it would be cheaper to `getJobs` in `mlApiServicesProvider` as it isn't calling additional datafeed and stats endpoints under the hood.

Reply:
I also need job and datafeed states. Shall I keep it as is then?

Reply:
ah ok, yes. I missed that it was being passed into `ConfigValidator` below